src/controlplane/discoverer/discoverer_test.go

```diff
@@ -0,0 +1,193 @@
+package discoverer_test
+
+import (
+	"context"
+	"testing"
+
+	"github.com/newrelic/infra-integrations-sdk/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/newrelic/nri-kubernetes/v2/internal/config"
+	"github.com/newrelic/nri-kubernetes/v2/internal/discovery"
+	"github.com/newrelic/nri-kubernetes/v2/src/controlplane/discoverer"
+)
+
+func Test_Discoverer_does_not_fail(t *testing.T) {
+	t.Parallel()
+
+	namespace := "testNamespace"
+	nodeName := "testNode"
+	selector := "foo=bar"
+
+	testCases := []struct {
```

Comment:
I might missed it, but I would write a tests and add a comments saying what happens if the list of selector provided is empty

Reply:
just added on "when_empty_selector", thanks